### PR TITLE
API tests: Use /api/endpoints

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Use `/api/endpoints` to verify everything is documented but also that there are not endpoints documented that do not exist. Before this PR, the tests only checked if all harmless (= `GET`) endpoints are there and whether their replied data matches the specified schema.

This PR adds testing if all documented API endpoints exist in FTL and if there are undocumented endpoints of *any* type (not limited to `GET`). This made us realize that the `/api/endpoints` reply itself needed a bit of updating to reflect the actual OpenAPI specification - more precisely, some `POST` do not require an `{item}` in their path as it is specified in the payload with `POST`.

The modified verification also unveiled that we didn't document the `GET /api/docs` endpoint and that a few `in: path` parameters were ambiguously named.

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Comment by DL6ER** [comment]
Example of what was returned before merging #1722:

```
Verifying the /api/endpoints endpoint...
  Errors:
  - Endpoint POST /action/reboot not found in FTL endpoints
  - Endpoint POST /action/poweroff not found in FTL endpoints
```

File: test/api/checkAPI.py [hunk 1]
```diff
@@ -48,39 +48,53 @@
...
+			if len(errors) == 0:
+				print("  GET " + path + " (" + verifyer.auth_method + " auth): OK")
+			else:
+				print("  GET " + path + " (" + verifyer.auth_method + " auth):")
+				for error in errors:
+					print("  - " + error)
+				errs[2] += len(errors)
+	print("")
+
+	# Verify that all the endpoint defined by GET /api/endpoints are documented
```
Comment (yubiuser) [inline test/api/checkAPI.py:72]:
```suggestion
	# Verify that all the endpoint defined by /api/endpoints are documented
```

**Review by yubiuser — COMMENTED** [review]
The example for `/docs` contains an empty selector box.

Screenshot at 2023-11-02 21-36-58

**Comment by DL6ER** [comment]
Screenshot from 2023-11-03 06-53-08
Screenshot from 2023-11-03 06-53-02

**Review by yubiuser — APPROVED** [review]

**Pull request merged**